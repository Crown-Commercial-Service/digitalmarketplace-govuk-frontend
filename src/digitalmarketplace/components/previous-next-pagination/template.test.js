/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('previous-next-pagination')

describe('Previous and Next Pagination', () => {
  describe('by default', () => {
    it('matches existing snapshot', () => {
      const $ = render('previous-next-pagination', examples.default)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('previous-next-pagination', examples.default, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('renders previous and next url and title', async () => {
      const $ = render('previous-next-pagination', examples.default, false, false, true)
      const $previous = $($('.dm-pagination__link')[0])
      const $next = $($('.dm-pagination__link')[1])

      expect($('.dm-pagination__link-label').length).toBe(0)

      expect($previous.find('.dm-pagination__link-text').text().trim()).toEqual('Previous page')
      expect($next.find('.dm-pagination__link-text').text().trim()).toEqual('Next page')
      expect($previous.attr('href')).toEqual('#')
      expect($next.attr('href')).toEqual('#')
    })
  })

  describe('with labels', () => {
    it('matches existing snapshot', () => {
      const $ = render('previous-next-pagination', examples['with labels'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders labels', async () => {
      const $ = render('previous-next-pagination', examples['with labels'], false, false, true)

      const $previous = $($('.dm-pagination__link-label')[0])
      const $next = $($('.dm-pagination__link-label')[1])
      expect($previous.text().trim()).toEqual('2 of 432')
      expect($next.text().trim()).toEqual('4 of 432')
    })
  })

  describe('with custom aria label', () => {
    it('matches existing snapshot', () => {
      const $ = render('previous-next-pagination', examples['with aria label'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders custom aria label', async () => {
      const $ = render('previous-next-pagination', examples['with aria label'], false, false, true)

      expect($('.dm-pagination').attr('aria-label')).toEqual('Custom aria label')
    })
  })

  describe('with no params', () => {
    it('matches existing snapshot', () => {
      const $ = render('previous-next-pagination', examples['with no params'])
      expect($.html()).toMatchSnapshot()
    })

    it('is empty', () => {
      const $ = render('previous-next-pagination', examples['with no params'])
      expect($.html()).toEqual('<html><head></head><body></body></html>')
    })
  })
})
