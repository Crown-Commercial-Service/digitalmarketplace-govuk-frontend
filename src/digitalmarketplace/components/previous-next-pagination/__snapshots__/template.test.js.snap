// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previous and Next Pagination by default matches existing snapshot 1`] = `
"<html><head></head><body><nav class=\\"dm-pagination\\" role=\\"navigation\\" aria-label=\\"Pagination\\">
    <ul class=\\"dm-pagination__list\\">
        <li class=\\"dm-pagination__item dm-pagination__item--previous\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"prev\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Previous page
              </span>
            </span>
          </a>
        </li> 
        <li class=\\"dm-pagination__item dm-pagination__item--next\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"next\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m10.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Next page
              </span>
            </span>
          </a>
        </li> 
    </ul>
 </nav>
</body></html>"
`;

exports[`Previous and Next Pagination with custom aria label matches existing snapshot 1`] = `
"<html><head></head><body><nav class=\\"dm-pagination\\" role=\\"navigation\\" aria-label=\\"Custom aria label\\">
    <ul class=\\"dm-pagination__list\\">
        <li class=\\"dm-pagination__item dm-pagination__item--previous\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"prev\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Previous page
              </span>
            </span>
              <span class=\\"dm-pagination__link-divider govuk-visually-hidden\\">:</span>
              <span class=\\"dm-pagination__link-label\\">2 of 432</span>
          </a>
        </li> 
        <li class=\\"dm-pagination__item dm-pagination__item--next\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"next\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m10.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Next page
              </span>
            </span>
              <span class=\\"dm-pagination__link-divider govuk-visually-hidden\\">:</span>
              <span class=\\"dm-pagination__link-label\\">4 of 432</span>
          </a>
        </li> 
    </ul>
 </nav>
</body></html>"
`;

exports[`Previous and Next Pagination with labels matches existing snapshot 1`] = `
"<html><head></head><body><nav class=\\"dm-pagination\\" role=\\"navigation\\" aria-label=\\"Pagination\\">
    <ul class=\\"dm-pagination__list\\">
        <li class=\\"dm-pagination__item dm-pagination__item--previous\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"prev\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Previous page
              </span>
            </span>
              <span class=\\"dm-pagination__link-divider govuk-visually-hidden\\">:</span>
              <span class=\\"dm-pagination__link-label\\">2 of 432</span>
          </a>
        </li> 
        <li class=\\"dm-pagination__item dm-pagination__item--next\\">
          <a href=\\"#\\" class=\\"dm-pagination__link\\" rel=\\"next\\">
            <span class=\\"dm-pagination__link-title\\">
              <svg class=\\"dm-pagination__link-icon\\" xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"13\\" width=\\"17\\" viewBox=\\"0 0 17 13\\">
                <path d=\\"m10.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z\\"></path>
              </svg>
              <span class=\\"dm-pagination__link-text\\">
                Next page
              </span>
            </span>
              <span class=\\"dm-pagination__link-divider govuk-visually-hidden\\">:</span>
              <span class=\\"dm-pagination__link-label\\">4 of 432</span>
          </a>
        </li> 
    </ul>
 </nav>
</body></html>"
`;

exports[`Previous and Next Pagination with no params matches existing snapshot 1`] = `"<html><head></head><body></body></html>"`;
